packages/statsd/src/client.ts

```diff
@@ -0,0 +1,127 @@
+import {StatsD, ClientOptions, Tags} from 'hot-shots';
+import {snakeCase} from 'change-case';
+
+export interface Timer {
+  stop(): number;
+}
+
+export interface Logger {
+  log(message: string): void;
+}
+
+export interface Options extends ClientOptions {
+  logger?: Logger;
+  snakeCase?: boolean;
+}
+
+export class StatsDClient {
+  private statsd: StatsD;
+  private logger: Logger | null = null;
+  private options: Options;
+
+  constructor({logger, ...options}: Options) {
+    this.logger = logger || null;
+    this.options = {
+      ...options,
+      errorHandler: options.errorHandler
+        ? options.errorHandler
+        : this.handleError.bind(this),
+    };
+
+    this.statsd = new StatsD(this.options);
+  }
+
+  distribution(stat: string | string[], value: number, tags?: Tags) {
+    return new Promise<void>((resolve, reject) => {
+      this.statsd.distribution(
+        stat,
+        value,
+        this.normalizeTags(tags),
+        this.createErrorCallback(resolve, reject),
+      );
+    });
+  }
+
+  increment(stat: string | string[], tags?: Tags) {
+    return new Promise<void>((resolve, reject) => {
+      this.statsd.increment(
+        stat,
+        1,
+        this.normalizeTags(tags),
+        this.createErrorCallback(resolve, reject),
+      );
+    });
+  }
+
+  close() {
+    return new Promise<void>((resolve, reject) => {
+      this.statsd.close(this.createErrorCallback(resolve, reject));
+    });
+  }
+
+  addGlobalTags(globalTags: Tags) {
+    this.statsd = this.statsd.childClient({
+      globalTags,
+    });
+  }
+
+  initTimer(): Timer {
+    const startTime = process.hrtime();
+
+    function stop() {
+      const duration = process.hrtime(startTime);
+      const [seconds, nanoseconds] = duration;
+      const milliseconds = seconds * 1000 + nanoseconds / 1e6;
+      return Math.round(milliseconds);
+    }
+
+    return {stop};
+  }
+
+  private handleError(error: Error) {
+    if (this.logger == null) {
+      return;
+    }
+
+    this.logger.log(
+      `Error occurred in the StatsD client:\n${error.stack || error.message}`,
+    );
+  }
+
+  private createErrorCallback<T>(
+    resolve: (value?: T | PromiseLike<T>) => void,
+    reject: (reason?: any) => void,
+  ) {
+    return (error?: Error) => {
+      if (error) {
+        if (this.options.errorHandler) {
+          this.options.errorHandler(error);
+        }
+        reject(error);
+      }
+
+      // Failing to send the metrics is not something we need to propagate
```

Comment:
This comment is a bit of a lie, as far as I can tell from the way this behaves / the code in general if we get here it is because there was no error (therefore sending stats succeeded). We can just🔥 this comment.